faust/serializers/codecs.py

```diff
@@ -358,11 +372,11 @@ def get_codec(name_or_codec: CodecArg) -> CodecT:
     return cast(Codec, name_or_codec)
 
 
-def dumps(codec: Optional[CodecArg], obj: Any) -> bytes:
+async def dumps(codec: Optional[CodecArg], obj: Any) -> bytes:
     """Encode object into bytes."""
-    return get_codec(codec).dumps(obj) if codec else obj
+    return await get_codec(codec).dumps(obj) if codec else obj
```

Comment:
I would split this out to a four-line `if ...: return else: return` construct. Await is new enough its precedence can be surprising. Are you awaiting the `get_codec(codec)`, the `get_codec(codec).dump(obj)` or the whole `if...else`?

Reply:
I am awaiting  `get_codec(codec).dump(obj)`. We can split it